lib/combine_pdf/renderer.rb

```diff
@@ -47,14 +47,14 @@ def object_to_pdf(object)
 
     def format_string_to_pdf(object)
       obj_bytes = object.bytes.to_a
-      # object.force_encoding(Encoding::ASCII_8BIT)
+      # object.dup.force_encoding(Encoding::ASCII_8BIT)
       if object.length == 0 || obj_bytes.min <= 31 || obj_bytes.max >= 127 # || (obj_bytes[0] != 68  object.match(/[^D\:\d\+\-Z\']/))
         # A hexadecimal string shall be written as a sequence of hexadecimal digits (0-9 and either A-F or a-f)
         # encoded as ASCII characters and enclosed within angle brackets (using LESS-THAN SIGN (3Ch) and GREATER- THAN SIGN (3Eh)).
-        "<#{object.unpack('H*')[0]}>".force_encoding(Encoding::ASCII_8BIT)
+        "<#{object.unpack('H*')[0]}>".dup.force_encoding(Encoding::ASCII_8BIT)
       else
         # a good fit for a Literal String or the string is a date (MUST be literal)
-        ('(' + ([].tap { |out| obj_bytes.each { |byte| out.concat(STRING_REPLACEMENT_ARRAY[byte]) } } ).pack('C*') + ')').force_encoding(Encoding::ASCII_8BIT)
+        ('(' + ([].tap { |out| obj_bytes.each { |byte| out.concat(STRING_REPLACEMENT_ARRAY[byte]) } } ).pack('C*') + ')').dup.force_encoding(Encoding::ASCII_8BIT)
```

Comment:
(again) This dup is redundant and will use up a bunch of memory. The dynamic String used here should be guaranteed to be mutable.